docs/resource/README.md

```diff
@@ -102,7 +102,105 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+and stay the same for the lifetime of the service instance, however it is acceptable that
+the ID is ephemeral and changes during important lifetime events for the service
+(e.g. service instance restarts).
+
+If the instance has no inherent unique ID that can be used as the value of this attribute,
+implementations MAY generate a random Version 1 or Version 4
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. When enough data is available,
+implementations SHOULD use Version 5 and SHOULD use the following UUID as the namespace:
+`4d63009a-8d0f-11ee-aad7-4c796ed8e320`.
+
+UUIDs are typically recommended, as only an opaque yet reproducible value for
+the purposes of identifying a service instance is needed. Similar to what can be seen in the man page for the
+[`/etc/machine-id`](https://www.freedesktop.org/software/systemd/man/machine-id.html) file,
+the underlying data, such as pod name and namespace should be treated as
+confidential by this algorithm, being the user's choice to expose it or not via another resource attribute.
+
+When a UUID v5 is generated, the input SHOULD be prefixed with
+`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}`, followed by the
+the workload identifier, which should tentatively be stable. The workload identifier is the final piece of
+information making the source of telemetry to be uniquely idenfiable. For instance, it might be the `container.name`
+in containerized or Kubernetes environments. This means that the same service yields the
+same UUID if the same identifier (`host.id`, `/etc/machine-id`, and so on) remains the same. It would still
+yield different results for different services on the same host or namespace. When no namespaces or equivalent fields
+are available, the prefix MUST then be `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`.
+
+SDKs SHOULD use the following algorithm when generating `service.instance.id`:
+
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this MUST take priority over generated IDs.
+- When the language, framework, or platform has a native mechanism allowing workloads to be uniquely idenfiable, like Erlang's
+  clustering identity, this mechanism MAY be used.
+- When any of the below combinations of resource attribute are provided, they MUST be used as the input
+  for generating a UUID v5 following the prefix mentioned above. The values within each combination MUST be separated with dots:
+  * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`, resulting in the input
+  `${telemetry.sdk.name}.${telemetry.sdk.language}.${k8s.namespace.name}.${service.name}.${k8s.pod.name}.${k8s.container.name}`. In this case,
+  the namespace MUST be used.
+  * `container.id`, resulting in the input `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${container.id}`,
```

Comment:
`service.name` can actually be missing effectively  (e.g. `unknown_service:java`) - I think we should skip these variants then

Reply:
Service name is a required field, "unknown_service" is an anomalous situation. IMO, we should count on those attributes being present.

Reply:
This is not the situation today - e.g. the javaagent uses unknown_service:java if it can't discover a name.

Reply:
+1 to treating `unknown_service:.*` as the same as empty.

Worst cast, we degrade down to raw UUIDs here.   `service.name` is something we should encourage all users to fill out *and make it easy* with things like the otel-operator.  However, degrading to UUID gives decent o11y behavior.  We almost want failure to provide a better `service.instance.id` to be a FAQ of 

> Q: "why am i getting this sub-par instance id"
> A: "Go configure service.name"

Reply:
I guess it boils down to the following question:

If, for whatever reason, we have no meaningful `service.name`, but have `host.id` (e.g. from `/etc/machine-id`) or `container.id`, we have those choices

-`${telemetry.sdk.name}.${telemetry.sdk.language}.${container.id}` - assuming that we probably only have a single service running on the host/container
- uuidv4 - assuming that the risk of a collision is too high

Reply:
If we assume that there's only one service running on a given container, there's no difference between using either version as the input for the UUID:

```
${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${container.id}
```
vs
```
${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${container.id}
```

Those two inputs have the same quality, given that the container-id _should_ be unique enough within a given namespace.

Reply:
> Worst cast, we degrade down to raw UUIDs here

@jsuereth are you saying that we should always fall back to UUID if the service.name is missing, even if we're running on a pod container?